src/Disks/IDisk.cpp

```diff
@@ -286,6 +288,22 @@ catch (Exception & e)
     throw;
 }
 
+void IDisk::checkCaseSensitivity()
+{
+    const String path = fmt::format("clickhouse_case_sensitivity_check_{}", toString(DB::UUIDHelpers::generateV4()));
```

Comment:
Maybe we can just use the same `checkAccessImpl` for it, to avoid creating extra file?

Reply:
Sometimes we skip check access, for example in `clickhouse-local` here:
https://github.com/ClickHouse/ClickHouse/blob/901b98d8563b873d178b2f658b1d578b4be41f21/programs/local/LocalServer.cpp#L622

But we need to check case sensitivity in `clickhouse-local` as well, because we can create `MergeTree` there.